src/main/java/net/fabricmc/loader/discovery/BuiltinMetadataWrapper.java

```diff
@@ -0,0 +1,98 @@
+
+import com.google.gson.JsonElement;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.loader.api.Version;
+import net.fabricmc.loader.api.metadata.ContactInformation;
+import net.fabricmc.loader.api.metadata.ModDependency;
+import net.fabricmc.loader.api.metadata.ModMetadata;
+import net.fabricmc.loader.api.metadata.Person;
+import net.fabricmc.loader.metadata.EntrypointMetadata;
+import net.fabricmc.loader.metadata.LoaderModMetadata;
+import net.fabricmc.loader.metadata.NestedJarEntry;
+
+class BuiltinMetadataWrapper implements LoaderModMetadata {
+	public BuiltinMetadataWrapper(ModMetadata parent) {
+		this.parent = parent;
+	}
+
+	@Override
+	public String getType() { return parent.getType(); }
+	@Override
+	public String getId() { return parent.getId(); }
+	@Override
+	public Version getVersion() { return parent.getVersion(); }
+	@Override
+	public Collection<ModDependency> getDepends() { return parent.getDepends(); }
+	@Override
+	public Collection<ModDependency> getRecommends() { return parent.getRecommends(); }
+	@Override
+	public Collection<ModDependency> getSuggests() { return parent.getSuggests(); }
+	@Override
+	public Collection<ModDependency> getConflicts() { return parent.getConflicts(); }
+	@Override
+	public Collection<ModDependency> getBreaks() { return parent.getBreaks(); }
+	@Override
+	public String getName() { return parent.getName(); }
+	@Override
+	public String getDescription() { return parent.getDescription(); }
+	@Override
+	public Collection<Person> getAuthors() { return parent.getAuthors(); }
+	@Override
+	public Collection<Person> getContributors() { return parent.getContributors(); }
+	@Override
+	public ContactInformation getContact() { return parent.getContact(); }
+	@Override
+	public Collection<String> getLicense() { return parent.getLicense(); }
+	@Override
+	public Optional<String> getIconPath(int size) { return parent.getIconPath(size); }
+	@Override
+	public boolean containsCustomElement(String key) { return parent.containsCustomElement(key); }
+	@Override
+	public JsonElement getCustomElement(String key) { return parent.getCustomElement(key); }
+	@Override
+	public int getSchemaVersion() { return Integer.MAX_VALUE; }
+	@Override
+	public Map<String, String> getLanguageAdapterDefinitions() { return Collections.emptyMap(); }
+	@Override
+	public Collection<NestedJarEntry> getJars() { return Collections.emptyList(); }
+	@Override
+	public Collection<String> getMixinConfigs(EnvType type) { return Collections.emptyList(); }
+	@Override
+	public boolean loadsInEnvironment(EnvType type) { return true; }
+	@Override
+	public Collection<String> getOldInitializers() { return Collections.emptyList(); }
+	@Override
+	public List<EntrypointMetadata> getEntrypoints(String type) { return Collections.emptyList(); }
+	@Override
+	public Collection<String> getEntrypointKeys() { return Collections.emptyList(); }
+	@Override
+	public void emitFormatWarnings(Logger logger) { }
+
+	private final ModMetadata parent;
```

Comment:
In all of the existing fabric projects we have fields at the top. Might be best to keep it consistent unless there is a reason.

Reply:
changed

Reply:
Weird. Who puts fields _at the bottom_?